lib/ui/router/branch_container.dart

```diff
@@ -0,0 +1,43 @@
+import 'package:collection/collection.dart';
+import 'package:flutter/material.dart';
+
+class BranchContainer extends StatelessWidget {
+  const BranchContainer({
+    super.key,
+    required this.currentIndex,
+    required this.children,
+  });
+
+  final int currentIndex;
+
+  final List<Widget> children;
+
+  @override
+  Widget build(BuildContext context) {
+    return Stack(
+      children: children.mapIndexed(
+        (
+          int index,
+          Widget navigator,
+        ) {
+          return Opacity(
+            opacity: index == currentIndex ? 1 : 0,
+            child: _branchNavigatorWrapper(index, navigator),
+          );
+        },
+      ).toList(),
```

Comment:
```
      children: [
        ...children.mapIndexed((index, navigator) => Opacity(
            // TODO: animation
            opacity: index == currentIndex ? 1 : 0,

            /// Avoid tap event
            child: IgnorePointer(
              ignoring: index != currentIndex,

              /// Avoid scroll event
              child: TickerMode(
                enabled: index == currentIndex,
                child: navigator,
              ),
            ),
          ),
        ),
      ],
```

@HironobuIga 
I thought that this kind of implementation would be better, what do you think?
Or it seems better to add doc, which is a widget for ignoring Events, to `_branchNavigatorWrapper` function.

Reply:
@koji-1009 
Thank you for your suggestion. It makes sense.
This container is stack of widgets, and it's probably possible to detect tap event in multiple widget.
I agree with you that we need to ignore any tab or drag event to non selected index screen.

Reply:
@koji-1009 
Ok, your code looks simpler. I'll adopt it :)

Reply:
Refactored :)
[refactor: change structure of branch_container widget](https://github.com/FlutterKaigi/conference-app-2023/pull/142/commits/0f1532be54d3b357cd80e9656e8f1928a2755bc2)